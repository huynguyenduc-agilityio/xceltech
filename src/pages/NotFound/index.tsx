import { Link, useLocation } from 'react-router-dom';

// Components
import { Button } from '@/components';

// Constants
import { ADMIN_PAGE, USER_PAGE } from '@/constants';

const NotFound = () => {
  const location = useLocation();
  const hrefDashboard = location.pathname.includes('admin')
    ? ADMIN_PAGE.DASHBOARD
    : USER_PAGE.DASHBOARD;

  return (
    <div className="h-screen mx-auto text-center">
      <div className="flex flex-col items-center justify-center h-full pt-20 pb-24">
        <p className="text-black-default text-2xl font-semibold">
          Oops! Page not found
        </p>
        <p className="text-primary text-lg font-medium max-w-md mt-2">
          The page you are looking for might have been removed or temporarily
          unavailable.
        </p>
        <Link to={hrefDashboard}>
          <Button className="px-6 py-3 mt-10 transition">
            Back to Dashboard
          </Button>
        </Link>
      </div>
    </div>
  );
};

export default NotFound;
