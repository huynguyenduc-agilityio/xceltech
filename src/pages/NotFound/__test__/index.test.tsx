import { render } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';

// Pages
import NotFound from '..';

describe('NotFound Page', () => {
  test('should match snapshot', () => {
    const view = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );

    expect(view).toMatchSnapshot();
  });

  test('renders NotFound component correctly', () => {
    const { getByText, getByRole } = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    expect(getByText(/Oops! Page not found/i)).toBeInTheDocument();
    expect(
      getByText(
        /The page you are looking for might have been removed or temporarily unavailable/i,
      ),
    ).toBeInTheDocument();
    expect(
      getByRole('button', { name: /Back to Dashboard/i }),
    ).toBeInTheDocument();
  });
});
