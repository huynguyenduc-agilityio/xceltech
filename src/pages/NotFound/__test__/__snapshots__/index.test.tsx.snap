// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen mx-auto text-center"
      >
        <div
          class="flex flex-col items-center justify-center h-full pt-20 pb-24"
        >
          <p
            class="text-black-default text-2xl font-semibold"
          >
            Oops! Page not found
          </p>
          <p
            class="text-primary text-lg font-medium max-w-md mt-2"
          >
            The page you are looking for might have been removed or temporarily unavailable.
          </p>
          <a
            data-discover="true"
            href="/user/dashboard"
          >
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white h-[70px] px-6 py-3 mt-10 transition"
            >
              Back to Dashboard
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen mx-auto text-center"
    >
      <div
        class="flex flex-col items-center justify-center h-full pt-20 pb-24"
      >
        <p
          class="text-black-default text-2xl font-semibold"
        >
          Oops! Page not found
        </p>
        <p
          class="text-primary text-lg font-medium max-w-md mt-2"
        >
          The page you are looking for might have been removed or temporarily unavailable.
        </p>
        <a
          data-discover="true"
          href="/user/dashboard"
        >
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white h-[70px] px-6 py-3 mt-10 transition"
          >
            Back to Dashboard
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
